prov/gni/test/rdm_sr.c

```diff
@@ -282,23 +283,22 @@ void do_send(int len)
 	sz = fi_recv(ep[1], target, len, rem_mr, gni_addr[0], source);
 	cr_assert_eq(sz, 0);
 
-	while ((ret = fi_cq_read(msg_cq[0], &cqe, 1)) == -FI_EAGAIN) {
-		pthread_yield();
-	}
-
-	cr_assert_eq(ret, 1);
-	cr_assert_eq((uint64_t)cqe.op_context, (uint64_t)target);
+	/* need to progress both CQs simultaneously for rendezvous */
```

Comment:
This section worries me. I'm not sure that this type of behavior needs to be known to application in order for things to work properly for the GNI provider. 

It's my understanding that both CQs have to be progressed for the operation to complete, not because we are checking to see that both events are complete. I don't believe any other providers require such operation with their implementations and this feels hacky. If both CQs need to be read to progress the state of the rendezvous engine, perhaps we should have an internal thread touching those structures and progressing them as needed.